src/compiler/utilities.ts

```diff
@@ -3093,11 +3093,13 @@ namespace ts {
      * Gets the effective type parameters. If the node was parsed in a
      * JavaScript file, gets the type parameters from the `@template` tag from JSDoc.
      */
-    export function getEffectiveTypeParameterDeclarations(node: DeclarationWithTypeParameters) {
-        return node.typeParameters || (isInJavaScriptFile(node) ? getJSDocTypeParameterDeclarations(node) : undefined);
+    export function getEffectiveTypeParameterDeclarations(node: DeclarationWithTypeParameters | JSDocTypedefTag) {

```

Comment:
there are a few other places that we get type parameter declarations of typedef and callback tags. They explicitly do *not* use `getEffectiveTypeParameterDeclarations` because it incorrectly (1) finds the host and (2) looks for `@template` in *any* jsdoc comment. Jsdoc type aliases only look for `@template` in their containing comment.

That said, it's probably the Right Thing to make `getEffectiveTypeParameterDeclarations` handle jsdoc type aliases, and have everybody call it.

Reply:
https://github.com/Microsoft/TypeScript/pull/23947 isn't merged yet

Reply:
> That said, it's probably the Right Thing to make getEffectiveTypeParameterDeclarations handle jsdoc type aliases, and have everybody call it.

Do you not have such a fix in #23947 ?

Reply:
Yes, along with the astonishing ability to forget what I did two weeks ago!